packages/realm-server/tests/permissions/permission-checker-test.ts

```diff
@@ -1,41 +1,118 @@
   module('world-readable realm', function () {
-    let permissionsChecker = new RealmPermissionChecker({
-      '*': ['read'],
-    });
+    let permissionsChecker = new RealmPermissionChecker(
+      {
+        '*': ['read'],
+      },
+      mockMatrixClient,
+    );
+
+    test('anyone can read but not write', async function (assert) {
+      assert.ok(await permissionsChecker.can('anyone', 'read'));
+      assert.notOk(await permissionsChecker.can('anyone', 'write'));
 
-    test('anyone can read but not write', function (assert) {
-      assert.ok(permissionsChecker.can('anyone', 'read'));
-      assert.notOk(permissionsChecker.can('anyone', 'write'));
+      assert.deepEqual(await permissionsChecker.for('anyone'), ['read']);
     });
   });
 
   module('world-writable realm', function () {
-    let permissionsChecker = new RealmPermissionChecker({
-      '*': ['read', 'write'],
+    let permissionsChecker = new RealmPermissionChecker(
+      {
+        '*': ['read', 'write'],
+      },
+      mockMatrixClient,
+    );
+
+    test('anyone can read and write', async function (assert) {
+      assert.ok(await permissionsChecker.can('anyone', 'read'));
+      assert.ok(await permissionsChecker.can('anyone', 'write'));
+
+      assert.deepEqual(await permissionsChecker.for('anyone'), [
+        'read',
+        'write',
+      ]);
     });
+  });
+
+  module('users-readable realm', function () {
+    let permissionsChecker = new RealmPermissionChecker(
+      {
+        users: ['read', 'write'],
+        '@matic:boxel-ai': ['read', 'write'],
+      },
+      mockMatrixClient,
+    );
+
+    test('matrix user can read and write', async function (assert) {
+      assert.ok(await permissionsChecker.can('@matic:boxel-ai', 'read'));
+      assert.ok(await permissionsChecker.can('@matic:boxel-ai', 'write'));
+
+      assert.deepEqual(await permissionsChecker.for('@matic:boxel-ai'), [
+        'read',
+        'write',
+      ]);
+
+      matrixUserProfile = { displayname: 'Not Matic' };
+      assert.ok(await permissionsChecker.can('@not-matic:boxel-ai', 'read'));
+      assert.ok(await permissionsChecker.can('@not-matic:boxel-ai', 'write'));
 
-    test('anyone can read and write', function (assert) {
-      assert.ok(permissionsChecker.can('anyone', 'read'));
-      assert.ok(permissionsChecker.can('anyone', 'write'));
+      assert.deepEqual(await permissionsChecker.for('@not-matic:boxel-ai'), [
+        'read',
+        'write',
+      ]);
+    });
+
+    test('non-matrix user can not read and write', async function (assert) {
+      assert.ok(await permissionsChecker.can('@matic:boxel-ai', 'read'));
+      assert.ok(await permissionsChecker.can('@matic:boxel-ai', 'write'));
+
+      assert.deepEqual(await permissionsChecker.for('@matic:boxel-ai'), [
+        'read',
+        'write',
+      ]);
+
+      matrixUserProfile = undefined;
+      assert.notOk(await permissionsChecker.can('@not-matic:boxel-ai', 'read'));
```

Comment:
can we use a different username for the non-matrix user? it's confusing because in some tests `@not-matic:boxel-ai` is a matrix user (I assume), and in this test it's not a matrix user